apps/chat-e2e/src/tests/promptFoldersSpecialChars.test.ts

```diff
@@ -0,0 +1,209 @@
+import dialTest from '@/src/core/dialFixtures';
+import {
+  ExpectedConstants,
+  ExpectedMessages,
+  MenuOptions,
+} from '@/src/testData';
+import { expect } from '@playwright/test';
+
+dialTest(
+  'Prompt folder: Error message appears if there is a dot is at the end of folder name.\n' +
+    'Prompt folder: allowed special characters.\n' +
+    'Prompt folder: restricted special characters are not entered.\n' +
+    'Prompt folder: restricted special characters are removed if to copy-paste.\n' +
+    'Prompt folder: spaces in the middle of folder name stay.\n' +
+    'Prompt folder: name can not be blank or with spaces only.\n' +
+    'Prompt folder: spaces at the beginning or end of folder name are removed.\n' +
+    'Prompt folder: smiles, hieroglyph, specific letters in name',
+  async ({
+    dialHomePage,
+    promptBar,
+    folderPrompts,
+    folderDropdownMenu,
+    errorToast,
+    setTestIds,
+    promptBarFolderAssertion,
+  }) => {
+    setTestIds(
+      'EPMRTC-2975',
+      'EPMRTC-2976',
+      'EPMRTC-2977',
+      'EPMRTC-2978',
+      'EPMRTC-2979',
+      'EPMRTC-2980',
+      'EPMRTC-2981',
+      'EPMRTC-2982',
+    );
+    const folderName = ExpectedConstants.newFolderWithIndexTitle(1);
+    const newNameWithEndDot = `${folderName}.`;
+    const newNameWithSpecialChars = `${folderName} ${ExpectedConstants.allowedSpecialChars}`;
+    const nameWithRestrictedChars = `Folder${ExpectedConstants.restrictedNameChars}_name`;
+    const expectedFolderName = 'Folder_name';
+    const newNameWithSpaces = 'Folder   1';
+    const expectedName = 'Folder with spaces';
+    const nameWithSpacesBeforeAndAfter = `   ${expectedName}   `;
+    const newNameWithEmojis = '😂👍🥳 😷 🤧 🤠 🥴😇 😈 ⭐あおㅁㄹñ¿äß';
+
+    await dialTest.step('Create prompt folder', async () => {
+      await dialHomePage.openHomePage();
+      await dialHomePage.waitForPageLoaded();
+
+      await promptBar.createNewFolder();
+      await promptBarFolderAssertion.assertFolderState(
+        { name: folderName },
+        'visible',
+      );
+    });
+
+    await dialTest.step(
+      'Rename it to have a dot at the end of the name',
+      async () => {
+        await folderPrompts.openFolderDropdownMenu(folderName);
+        await folderDropdownMenu.selectMenuOption(MenuOptions.rename);
+        await folderPrompts.editFolderName(newNameWithEndDot);
+      },
+    );
+
+    await dialTest.step('Click on confirmation button', async () => {
+      await folderPrompts.getEditFolderInputActions().clickTickButton();
+
+      const errorMessage = await errorToast.getElementContent();
+      expect
+        .soft(errorMessage, ExpectedMessages.notAllowedNameErrorShown)
+        .toBe(ExpectedConstants.nameWithDotErrorMessage);
+
+      // Verify folder name stays in edit mode
+      await expect
```

Comment:
let's move this verification to `FolderAssertion`  to have smth like:

`await promptBarFolderAssertion.assertFolderEditInputState('visible');`